api/bots_api/bots_test_lib.py

```diff
@@ -35,31 +39,32 @@ def check_expected_responses(self, expectations, expected_method='send_reply', e
             if type != "stream":
                 self.mock_test(
                     {'content': m, 'type': "private", 'display_recipient': recipient,
-                     'sender_email': email}, r, expected_method)
+                     'sender_email': email}, r, expected_method, http_request, http_response)
             if type != "private":
                 self.mock_test(
                     {'content': m, 'type': "stream", 'display_recipient': recipient,
-                     'subject': subject, 'sender_email': email}, r, expected_method)
+                     'subject': subject, 'sender_email': email}, r, expected_method, http_request, http_response)
 
-    def mock_test(self, messages, bot_response, expected_method):
-        # type: (Dict[str, str], Any, str) -> None
+    def mock_test(self, messages, bot_response, expected_method, http_request=None, http_response=None):
+        # type: (Dict[str, str], Any, str, Dict[str, Any], Dict[str, Any]) -> None
         if expected_method == "send_reply":
-            self.mock_test_send_reply(messages, bot_response, expected_method)
+            self.mock_test_send_reply(messages, bot_response, expected_method, http_request, http_response)
         else:
-            self.mock_test_send_message(messages, bot_response, expected_method)
+            self.mock_test_send_message(messages, bot_response, expected_method, http_request, http_response)
```

Comment:
For arguments that are accepted as keyword arguments, we like to pass them as keyword arguments as well.  It's  generally bad style (and often leads to bugs) to just use the fact that the 4th positional argument, `http_response`, is here.

I would also consider getting rid of the `mock_test_send_message` and `mock_test_send_reply` helpers and just inline their code here; those functions have more code related to passing arguments around than actually doing anything.